microbootstrap/bootstrappers/base.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+import dataclasses
+import typing
+
+from microbootstrap.helpers import dataclass_to_dict_no_defaults, merge_dataclasses_configs, merge_dict_configs
+from microbootstrap.instruments import OpentelemetryInstrumentConfig, SentryInstrumentConfig
+
+
+if typing.TYPE_CHECKING:
+    from microbootstrap.instruments import Instrument
+    from microbootstrap.settings.base import BootstrapSettings
+
+
+ApplicationT = typing.TypeVar("ApplicationT")
+SettingsT = typing.TypeVar("SettingsT", bound="BootstrapSettings")
+SelfT = typing.TypeVar("SelfT", bound="ApplicationBootstrapper[typing.Any, typing.Any, typing.Any]")
```

Comment:
Not sure if it’s necessary

Reply:
In python 3.9 there is no such convenient way like typing.Self from future versions
And if we get rid of this SelfT then methods would look like this
```python
    def configure_opentelemetry(
        self: ApplicationBootstrapper[typing.Any, typing.Any, typing.Any],
        opentelemetry_config: OpentelemetryInstrumentConfig,
    ) -> ApplicationBootstrapper[typing.Any, typing.Any, typing.Any]:
        self.opentelemetry_instrument.configure_instrument(opentelemetry_config)
        return self
```

Ugly a bit imo

Reply:
Yes, quite ugly. I prefer to avoid annotating `self` at all. `mypy` is happy this way too

Reply:
But we need the completion hints to indicate that we are returning the same object on which the method was called.
This looks normal, maybe
```python
    def configure_opentelemetry(
        self,
        opentelemetry_config: OpentelemetryInstrumentConfig,
    ) -> ApplicationBootstrapper[typing.Any, typing.Any, typing.Any]:
        self.opentelemetry_instrument.configure_instrument(opentelemetry_config)
        return self
```
But it's okay to use SelfT, looks more natural, i guess

Reply:
Oh, I get it now. What do you think about adding typing-extensions to dependencies?

Reply:
Yeah, 100% should do it